### ISSUE TITLE
API Endpoint

### ISSUE THREAD

**Issue by ltoshea** [report-uri/report-uri-bugs#69]
I'd love to be able to pull the CSP raw report info column via an API.  As it is I can see that it's doing an AJAX call but the response is going to be quite nasty to parse. Do you offer an API endpoints or ways to get the report information programatically? 
Thanks

**Comment by hackjammer** [comment]
@ScottHelme any thoughts on this?

**Comment by ScottHelme** [comment]
An API for various features is definitely on the cards. For this particular request, what would you like to query on? My guess would be the time window, like a particular day/hour/month, to return all the reports. With that you'd just like the raw JSON?

**Comment by hackjammer** [comment]
Yeah that would be perfect - raw JSON for a time period would be enough for us for this

**Comment by ltoshea** [comment]
I second this, thanks

**Comment by ltoshea** [comment]
@ScottHelme do you have a rough time frame for this -are we talking weeks or months?
Thanks

**Comment by ltoshea** [comment]
Hi Scott, do you have a better idea on the time frame of this feature?
Thanks

**Comment by anroots-tw** [comment]
Also looking for an API/export functionality. :+1: 

This would help us with doing more complex reporting/analysis if we wanted to as well as piping the data to other log/analytics tools.

**Comment by lionelandre** [comment]
+1 on this as it would potentially allow us to feed splunk to trigger alerts if new entries are found in a report.  Our expectation is that once CSP is in place, block reports should be few and far between....

**Comment by stefanhorning** [comment]
Same here, at least provide metrics (counts per time window) for futher alerting etc.

**Comment by kurtfoster** [comment]
+1 on this feature
I'd also love an API for configuration. Mainly the list of domains that we accept reports from, but any other features as well. We could then integrate this directly into our CMDB rather than manually maintaining config.

**Comment by omerlh** [comment]
Also looking for either API or webhook we can use for monitoring (e.g. Prometheus exporter or Slack message on each new report)

**Comment by eloquence** [comment]
We'd love to integrate alerting for CSP e.g. with our icinga setup. In lieu of an API, is there a recommended method for getting real-time alerts about CSP issues?
